packages/approval-controller/src/ApprovalController.ts

```diff
@@ -90,8 +91,10 @@ export type ApprovalRequest<RequestData extends ApprovalRequestData> = {
 
   /**
    * The type of the approval request.
+   * Unfortunately, not all values will match the `ApprovalType` enum.
+   * The `string` type would ideally be removed one day.
```

Comment:
Let's make this a todo:

```suggestion
   * TODO: Remove the `string` member by either narrowing the `type` property, or expanding the `ApprovalType` enum, to encompass all approval request types used in the clients.
```

I think this would be a sensible way to achieve sound typing in the future:

> We could push for every type to be defined here first then make this change, but it forces an awkward workflow as new confirmations need to be added to the enum before they are implemented.

https://github.com/MetaMask/core/pull/4209#discussion_r1577942472